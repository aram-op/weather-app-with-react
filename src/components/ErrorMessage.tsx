import '../styles/error-message.css';

function ErrorMessage({message} : {message: string}) {
    return (
        <h2 className="error-message">{message}</h2>
    );
}

export default ErrorMessage;