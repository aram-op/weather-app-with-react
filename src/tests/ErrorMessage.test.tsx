import React from 'react';
import { render, screen } from '@testing-library/react';
import ErrorMessage from '../components/ErrorMessage';

describe('ErrorMessage', () => {

    it('should render the correct error message', () => {
        const message = 'Something bad happened :(';

        render(<ErrorMessage message={message} />);

        const errorMessageElement = screen.getByText(message);

        expect(errorMessageElement).toBeInTheDocument();
    });

    it('should match snapshot', () => {
        const message = 'Terrifying error occurred';

        const { asFragment } = render(<ErrorMessage message={message} />);

        expect(asFragment()).toMatchSnapshot();
    });
});
