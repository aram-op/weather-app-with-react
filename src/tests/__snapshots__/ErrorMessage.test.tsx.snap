// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <h2
    class="error-message"
  >
    Terrifying error occurred
  </h2>
</DocumentFragment>
`;

exports[`ErrorMessage should match snapshot 1`] = `
<DocumentFragment>
  <h2
    class="error-message"
  >
    Terrifying error occurred
  </h2>
</DocumentFragment>
`;
